### ISSUE TITLE
GetMapObjects only working once or twice

### ISSUE THREAD

**Issue by robbederks** [FeroxRev/Pokemon-Go-Rocket-API#49]
Hi!

I am trying to perform a few (let's say 10) GetMapObjects requests in a row. Between each request, I'm changing the location by around 50m.

This works for one or sometimes two requests, but then it just times out with the error:   

`Error: System.Threading.Tasks.TaskCanceledException: A task was canceled.`

This even happens if I wait for 5 seconds between requests, which is considered the minimum amount of time to wait.

Any clue why?
Thanks!

**Comment by unnown** [comment]
could be a softban, could be that you aren't updating your location towards the new points?

**Comment by robbederks** [comment]
Hmm...
Seems strange, because the first API call is always working. Even if I quit the program and reopen it immediately, the call works again.

I'll check on my new-location code. Maybe the positions are too far apart?

EDIT:
Even if I try to check the same location again, it times out...
Could it be possible that I need to call other functions of the API before Niantic lets me use the GetMapObjects again?

**Comment by unnown** [comment]
it could, but you always need to update your location anyway, so there already is a call in between them

**Comment by robbederks** [comment]
I found one error in my code. I was calling the client.Player.SetCoordinates function, but that doesn't make a request. Maybe it's a good idea to make this private if?

Changed that, but still no luck.

Now it seems like after I log in, only one request at all can be sent. I update the location the first time, after that, the GetMapObjects already times out.

I'll try logging in to the official app to see if it still works and I'm not soft-banned :-)

EDIT:
Doesn't seem like it, still functions like normal :-/

**Comment by unnown** [comment]
indeed always use:
var result = await _client.Player.UpdatePlayerLocation(waypoint.Latitude, waypoint.Longitude, _client.Settings.DefaultAltitude);

perhaps find/take part of the humanlikewalking code used in several bots?

**Comment by robbederks** [comment]
I found the problem, and it's not caused by your awesome library.

I'm using the API in combination with Xamarin on Android, and apparently the standard HTTPMessageHandler is pretty shit.
I switched over to the ModernHttpClient-handler, and now everything works without a problem :-)

Thank you anyway!

**Comment by FeroxRev** [comment]
Cool!

**Comment by echo000** [comment]
I know I'm a bit late to this party, but could you explain what you did to fix it @robbederks ?

**Comment by robbederks** [comment]
Of course, 

I'm using the Xamarin framework to create an Android app written in C#. I have recompiled the POGOProto and Protolib because it wasn't available for Mono.Android, and included this API. 

You use the HTTPMessageHandler in the base RPC, but that is pretty buggy in Mono.Android. This is the reason for only a few successful requests before it hangs. Therefor, I replaced it with the 3rd party library ModernHTTPClient -  handler, and now it works perfectly :-) 

This should only be an issue for Mono.Android and possibly Mono.iOS users though.
